### PR TITLE
Support Kubernetes

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by johrstrom** [comment]
Let's simplify the whole "data options for cluster" to just be kubernetes. If a module/option exists on both pizer & owens we'll support it in kubernetes - otherwise we won't.  I'll bring this up in our classroom meetings and we'll either support this scheme or backport the modules to owens.

File: form.yml.erb [hunk 1]
```diff
@@ -7,6 +7,9 @@
 cluster:
   - "owens"
   - "pitzer"
+<% if !OodAppkit.clusters["kubernetes"].nil? && OodAppkit.clusters["kubernetes"].allow? -%>
+  - "kubernetes"
+<% end -%>
```
Comment (johrstrom) [inline form.yml.erb:12]: I know you left a comment on jupyter, but we shouldn't need this. Also - I think they're symbols not strings, just FYI.

Comment (treydock) [reply form.yml.erb]: I removed the ACL checks, and it was working with stings for `kubernetes` key though symbols may have also worked

**Comment by treydock** [comment]
I made it so the only RStudio Server versions available for `kubernetes` are ones where they exist for both Pitzer and Owens.  That removed a lot of code changes, especially to the `form.js`.

File: template/after.sh.erb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+<% if context.cluster != 'kubernetes' -%>
```
Comment (johrstrom) [inline template/after.sh.erb:1]: I think this is now `/kubernetes.*/.match?(context.cluster)` with the test/dev clusters.

Comment (johrstrom) [reply template/after.sh.erb:1]: There are other statements like this in other files too.

**Comment by treydock** [comment]
Should I go ahead and add `kubernetes-test` and `kubernetes-dev` clusters to form list? They won't show up if they aren't defined so maybe keep things simpler for people doing app testing and what not so we don't have to maintain a bunch of forks and branches and what not?

File: template/after.sh.erb (hunk 2 above)
Comment (treydock) [reply template/after.sh.erb]: Fixed to use regex match for `/kubernetes/` where appropriate

**Comment by johrstrom** [comment]
Yea that'll be fine.

**Comment by treydock** [comment]
Updated with commits so should still work on dev and test instances of our ondemand servers.

**Comment by treydock** [comment]
Testing in test , I missed a few places in Javascript to handle test/dev names.

**Comment by treydock** [comment]
Fix test/dev case to also work, just had to update form fields to account for the 2 non-production Kubernetes cluster names.

File: form.yml.erb [hunk 3]
```diff
@@ -135,52 +197,82 @@ attributes:
       - [
           "4.0.2", "app_rstudio_server/4.0.2",
           data-option-for-owens: true,
-          data-option-for-pitzer: true
+          data-option-for-pitzer: true,
+          data-option-for-kubernetes: true,,
```
Comment (johrstrom) [inline form.yml.erb:201]: I think these extra commas are throwing the yaml parsing off?

Comment (treydock) [reply form.yml.erb]: I fixed, was likely existing trailing comma and I was copy+paste the items and must have missed.

**Review by johrstrom — APPROVED** [review]
I think we're good to go.

**Pull request merged**